src/graph/graph_op.cc

```diff
@@ -273,4 +297,22 @@ IdArray GraphOp::ExpandIds(IdArray ids, IdArray offset) {
   return rst;
 }
 
+ImmutableGraph GraphOp::ToSimpleGraph(const GraphInterface* graph) {
+  std::vector<dgl_id_t> indptr(graph->NumVertices() + 1), indices;
+  indptr[0] = 0;
+  for (dgl_id_t src = 0; src < graph->NumVertices(); ++src) {
+    std::unordered_set<dgl_id_t> hashmap;
+    for (const dgl_id_t dst : graph->SuccVec(src)) {
+      if (!hashmap.count(dst)) {
+        indices.push_back(dst);
+        hashmap.insert(dst);
+      }
+    }
+    indptr[src+1] = indices.size();
+  }
+  CSRPtr csr(new CSR(graph->NumVertices(), indices.size(),
+        indptr.begin(), indices.begin(), RangeIter(0), false));
+  return ImmutableGraph(csr);
```

Comment:
if the original graph index has only in_csr. the code first generates out_csr and return out_csr. Later on, the user's code needs in_csr. Maybe we don't need to spend time optimizing this operation. I just raise a potential issue. It's up to you to decide.

Reply:
Actually `SuccVec` will create `out_csr`, but you are right, it is always a decision to create which csr. I don't have good solution. It looks like a layout optimization problem that requires more information than simply one operator. For example, if we have a computation graph, then we know when and where to use which format so the format conversion cost + computation cost is minimized. Pretty far goal I will say.